### ISSUE TITLE
Custom attribute of customer is not save without cache flush.

### ISSUE THREAD

**Issue by vishalgchetu** [magento/magento2#20351]
### Preconditions (*)

1. v2.3.0

### Steps to reproduce (*)

I am also not able to save attribute without cache flush .
$customer = $this->customerFactory->create()->load($customerId);
$customer->setLengthTimeIndustry($timeSpent);
```
try{
$customer->save();
}
catch (Exception $e) {
$message = __('Invalid Parameter.');
$this->messageManager->addError($message);
}
```
When I clear cache or flush cache then it works

### Expected result (*)

It should save without cache flush.

### Actual result (*)

```
1.Not able to save without cache flush.
```

**Comment by magento-engcom-team** [comment]
Hi @vishalgchetu. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [ ] Summary of the issue
- [ ] Information on your environment
- [ ] Steps to reproduce
- [ ] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me $VERSION instance` 

where `$VERSION` is [version tags](https://github.com/magento/magento2/tags) (starting from 2.2.0+) or develop branches (for example: 2.3-develop). 
For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@vishalgchetu do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [ ] yes
- [ ] no

**Comment by magento-engcom-team** [comment]
Hi @irajneeshgupta. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [ ] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [ ] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [ ] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [ ] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [ ] 6. Add label `Issue: Confirmed` once verification is complete. 

- [ ] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by irajneeshgupta** [comment]
@vishalgchetu 
What is Your Magento version (are you able to recreate in vanilla instance)?  
Can you please describe your issue as per git format?
Preconditions, Steps to reproduce, Expected result, Actual result

**Comment by vishalgchetu** [comment]
I am using Magento 2.1.5 on production , we update one customer attribute after login as per API response  , sometimes all customer custom attribute data lost that is also a issue of magento , when we try to update all customer custom  attribute ,  it is not updating  but when we flush cache , it work . 

Its my production issue Can  you please look into this also we why customer   custom attribute data are  lost during customer save .  My application is syncing with Oracle database  , due to we  restore customer information Oracle database  when customer information lost  , but this is not good thing .

Thanks

**Comment by magento-engcom-team** [comment]
@irajneeshgupta Thank you for verifying the issue. Based on the provided information internal tickets `MAGETWO-97681` were created

**Comment by magento-engcom-team** [comment]
Hi @engcom-backlog-nazar. Thank you for working on this issue. 
Looks like this issue is already verified and confirmed. But if your want to validate it one more time, please, go though the following instruction:
- [ ] 1. Add/Edit `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.

- [ ] 2. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 3. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [ ] 4. If the issue is not relevant or is not reproducible any more, feel free to close it.

**Comment by ghost** [comment]
Hi @vishalgchetu thank you for you report, this issue has already fixed in 2.3-develop branch and will be available on 2.3.1 release.

**Comment by vishalgchetu** [comment]
Hi @engcom-backlog-nazar  : Can you provide me patch for this issue .
